### ISSUE TITLE
Impossible to define a "true" label for externally imported files ?

### ISSUE THREAD

**Issue by Overdrivr** [gpoore/minted#146]
I wish to import a source file using `\inputminted` and in the same time define a true label that can be used for cross-referencing. Documentation indicates that the `listing` environment should be used, but I am not sure both are compatible.

> label (string) (default: empty )
Add a label to the top, the bottom or both of the frames around the code. See the
fancyvrb documentation for more information and examples. **Note: This does not
add a \label to the current listing. To achieve that, use a floating environment
(section 4) instead.**

Is this possible ?

EDIT:

Also, if external files cannot be imported in the floated environment, I understand they cannot be referenced inside the list of listings, which I would need as well.

> The \listoflistings macro will insert a list of all (floated) listings in the document:

**Comment by gpoore** [comment]
The `label` argument is really a title, but the name was chosen by `fancyvrb`, so `minted` carries it over, since `minted` uses `fancyvrb` to do the code typesetting.

For a cross reference, you probably want something like this:
```latex
\begin{listing}
\inputminted{<language>}{<file>}
\label{<label>}
\end{listing}
```

**Comment by Overdrivr** [comment]
Yes that's exactly the snippet I'm looking for, sadly I already tried without success. The code snippet doesnt show up and is still not referenced inside the list of listings.

Any idea why it's not working ?

**Comment by gpoore** [comment]
It's working for me. And it shows up in the list of listings if I add a `\caption`.

Are you compiling enough times for the references to be resolved? Otherwise, what other packages are you loading? Perhaps there is a package conflict.

**Comment by Overdrivr** [comment]
Ok so it's partially working now, thanks a lot for the feedback !

I cannot reliably get the code to locate on the next page after the inclusion despite using the `!h` option.

But the main issue is that the code gets truncated, it does not fit in a single page and doesn't continue on the next one.

 ```
\begin{listing}[!hp]
\inputminted[frame=single]{VHDL}{src/2/snippets/tline.vhdl}
\label{foobar}
\caption{foo}
\end{listing}
```

Package list

```
% Page creation engine
\documentclass[11pt,a4paper]{book}
\usepackage[utf8x]{inputenc}
\usepackage[english]{babel}
\usepackage[T1]{fontenc}

% Main modules
\usepackage{url}      % URL
\usepackage{hyperref} % Links
\usepackage{graphicx} % Pictures
\usepackage{wrapfig}  % In-line Pictures
\usepackage{verbatim} % Code
\usepackage{moreverb} % Encoding
%\usepackage{listing}
\usepackage{minted}   % Code
\usepackage{color}    % Highlight
\usepackage{xcolor}   % Color text
\usepackage{textgreek}% Greek language
\usepackage{booktabs} % Tables
\usepackage[table,xcdraw]{xcolor} % For colors in tables
\usepackage{colortbl} % For coloring rows, columns of table
\usepackage[toc,page]{appendix} % For managing appendix sections cleanly
\usepackage{algpseudocode} % For displaying pseudo-code
\usepackage{amsmath,empheq} % Math equations

% Glossary
\usepackage[nomain,acronym,xindy,toc]{glossaries}
\loadglsentries{./src/glos}
\makeglossaries
\usepackage{makeidx}
\makeindex

% Front page
\usepackage[ED=GEET-Elec, Ets=UT3]{tlsflyleaf}
```

And a screenshot to illustrate.

![image](https://cloud.githubusercontent.com/assets/1294805/21817721/689b0892-d765-11e6-98b0-4b070c402bfc.png)

Thanks again for your help

**Comment by gpoore** [comment]
In that case, the built-in `listing` environment won't do what you need.  From the documentation FAQ:
> **I need a listing environment that supports page breaks.** The built-in
listing environment is a standard float; it doesn’t support page breaks. You
will probably want to define a new environment for long floats. For example,
> ```latex
> \usepackage{caption}
> \newenvironment{longlisting}{\captionsetup{type=listing}}{}
> ```
> With the `caption` package, it is best to use `minted`’s `newfloat` package option.
> See http://tex.stackexchange.com/a/53540/10742 for more on listing
environments with page breaks.

You might also load `hyperref` last (sometimes there can be problems otherwise), remove `moreverb` unless it is actually used, and only load either `color` or `xcolor` (and only once). Otherwise, I don't see anything obvious in the preamble that might cause issues.

**Comment by Overdrivr** [comment]
Sorry, I missed that in the docs. The `caption` package is exactly what I needed and all my problems are now solved. Thanks a lot !
